web-app/src/components/DipMap.js

```diff
@@ -745,6 +745,186 @@ export default class DipMap extends React.Component {
+			});
+			startingprovs.push({
+				name: "pic",
+				favours: "France",
+				influencedby: { par: "France", bre: "France", bel: "Neutral" }, // belgium neutral
+			});
+			startingprovs.push({
+				name: "tyr",
+				favours: "Austria",
+				influencedby: {
+					mun: "Germany",
+					ven: "Italy",
+					tri: "Austria",
+					vie: "Austria",
+				},
+			});
+			startingprovs.push({
+				name: "boh",
+				favours: "Austria",
+				influencedby: { mun: "Germany", vie: "Austria" },
+			});
+			startingprovs.push({
+				name: "gal",
+				favours: "Austria",
+				influencedby: {
+					bud: "Austria",
+					vie: "Austria",
+					war: "Russia",
+					rum: "Neutral",
+				}, //and rumania
+			});
+			startingprovs.push({
+				name: "ukr",
+				favours: "Russia",
+				influencedby: {
+					war: "Russia",
+					mos: "Russia",
+					stp: "Russia",
+					rum: "Neutral",
+				},
+			});
+			startingprovs.push({
+				name: "fin",
+				favours: "Russia",
+				influencedby: { stp: "Russia", swe: "Neutral", nwy: "Neutral" },
+			});
+
+			startingprovs.push({
+				name: "ruh",
+				favours: "Germany",
+				influencedby: {
+					kie: "Germany",
+					mun: "Germany",
+					bel: "Neutral",
+					hol: "Neutral",
+				}, //and belgium and holland
+			});
+			startingprovs.push({
+				name: "sil",
+				favours: "Germany",
+				influencedby: { ber: "Germany", mun: "Germany", war: "Russia" },
+			});
+			startingprovs.push({
+				name: "pru",
+				favours: "Germany",
+				influencedby: { ber: "Germany", war: "Russia" },
+			});
+			startingprovs.push({
+				name: "arm",
+				favours: "Turkey",
+				influencedby: { ank: "Turkey", smy: "Turkey", sev: "Russia" },
+			});
+
+			//Here we check each non-SC and non-Sea territory. If all surrounding SCs are of the same power and none is "Neutral", colour them that power.
+			//Get all nodes, disqualify Sea and SC, and per node collect the edges in an array.
+			for (let prov in this.state.variant.Properties.Graph.Nodes) {
+				const node = this.state.variant.Properties.Graph.Nodes[prov];
+				if (!node.SC && node.Subs[""].Flags.Land) {
+					const borderprovs = [];
+					for (let edge in node.Subs[""].Edges) {
+						const edgenode = this.state.variant.Properties.Graph.Nodes[edge];
+						if (typeof edgenode !== "undefined") {
+							if (edgenode.SC) {
+								borderprovs.push(SCs[edgenode.Name]);
+							}
+						}
+					}
+
+					//check if all members of array are equal/not neutral.
+					let shoulddraw = true;
+					for (let i = 0; i < borderprovs.length; i++) {
+						if (
+							borderprovs[i] !== borderprovs[0] ||
+							borderprovs[i] == "Neutral"
+						) {
+							shoulddraw = false;
+							break;
+						}
+					}
+
+					//if all equal, draw the colour.
+					if (shoulddraw) {
+						const col = helpers.natCol(borderprovs[0], this.state.variant);
+						this.map.colorProvince(prov, col);
+					}
+
+					//if by the default rule we don't draw it, check if "special" rules apply.
+					else {
```

Comment:
It might be nicer to restructure this logic so that we populate `shouldDraw` correctly (taking into account special cases from the start position). Then we only need one call to`colorProvince`. The complex logic below could go into a function.

Reply:
Dear god probably yes. 
The code I'm writing is horrific. The way I get about to that code is even worse. 

It's like giving a toddler crayons and then letting him create an architecture building plan... :). 
I just wanted the functionality so I started adding, but please help in making the 'best practices'.